rust-connectors/sinks/kafka/src/main.rs

```diff
@@ -18,15 +24,49 @@ async fn main() -> anyhow::Result<()> {
         println!("{}", serde_json::to_string_pretty(&schema).unwrap());
         return Ok(());
     }
-    let opts: KafkaOpt = KafkaOpt::from_args();
-    opts.common.enable_logging();
+    let raw_opts = KafkaOpt::from_args();
+    raw_opts.common.enable_logging();
     info!(
         connector_version = env!("CARGO_PKG_VERSION"),
         git_hash = git_hash_version(),
         "Starting Kafka sink connector",
     );
 
-    opts.execute().await?;
+    // =======================================================
+    // Setup objects and actors based on opts
+    // =======================================================
+
+    let kafka_sink_deps = KafkaSinkDependencies::from_kafka_opt(raw_opts).await?;
+    let kafka_producer = &kafka_sink_deps.kafka_producer;
+    let kafka_partition = &kafka_sink_deps.kafka_partition;
+    let kafka_topic = &kafka_sink_deps.kafka_topic;
+
+    // =======================================================
+    // Streaming sequence starts here
+    // =======================================================
+
+    info!("Starting stream");
+
+    let mut stream = kafka_sink_deps
+        .common_connector_opt
+        .create_consumer_stream()
+        .await?;
+
+    while let Some(Ok(record)) = stream.next().await {
+        let mut kafka_record = FutureRecord::to(kafka_topic.as_str())
+            .payload(record.value())
+            .key(record.key().unwrap_or(&[]));
+        if let Some(kafka_partition) = &(kafka_partition) {
+            kafka_record = kafka_record.partition(*kafka_partition);
+        }
+
+        let enqueue_res = kafka_producer.send_result(kafka_record);
+
+        if let Err((error, _)) = enqueue_res {
+            error!("Kafka produce {:?}", error);
```

Comment:
I think we should stop processing here (ideally with printing the last read offset) instead of having gaps in the sent records sequence.

Reply:
Ok. Changing it. Adding "last_record_read" in the form of json line so that it is machine readable.
Also, what happens to connectors that exit?